jctools-core/src/main/java/org/jctools/queues/atomic/BaseLinkedAtomicQueue.java

```diff
@@ -53,6 +53,51 @@ protected final void spConsumerNode(LinkedQueueAtomicNode<E> node) {
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This method is only safe to call from the consumer thread!
+     */
+    @Override
+    public final boolean remove(Object o) {
+        if (null == o) {
```

Comment:
I used NPE because other methods which can throw or return `false` (e.g. `offer`) also take the "throw" approach. Both seem legit and I'm fine with which ever you prefer https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#remove-java.lang.Object-. FWIW I would prefer to avoid throwing and just return false too.